lib/datadog/grok/src/filters/array.rs

```diff
@@ -0,0 +1,187 @@
+use crate::{
+    ast::{Function, FunctionArgument},
+    parse_grok_rules::Error as GrokStaticError,
+};
+use bytes::Bytes;
+use nom::{
+    branch::alt,
+    bytes::complete::{tag, take, take_until},
+    character::complete::char,
+    combinator::cut,
+    combinator::map,
+    multi::separated_list0,
+    sequence::{preceded, terminated},
+    IResult,
+};
+use std::convert::TryFrom;
+
+use crate::grok_filter::GrokFilter;
+use vrl_compiler::Value;
+
+pub fn filter_from_function(f: &Function) -> Result<GrokFilter, GrokStaticError> {
+    let args_len = f.args.as_ref().map_or(0, |args| args.len());
```

Comment:
Could `f.args.as_ref()` be stored in a variable to save us from having to create a new ref every time?

Reply:
Good idea, done.